doc/decisions/operation_sequences.md

```diff
@@ -0,0 +1,92 @@
+# KDB operation sequences
+
+## Problem
+
+The current public API allows various kind of sequences of `kdbGet` and `kdbSet` calls.
+There is no real guideline of allowed and/or prohibited sequences, other than `kdbGet` for a certain parent key must be called before `kdbSet`:
+
+- initially after `kdbOpen`
+- after conflict errors in `kdbSet`
+
+While researching ways to do change tracking, we discovered that some sequences are currently legal, but might be problematic.
+This issue not only effects change tracking, but may cause problems for other use cases too.
+
+```c
+Key * keyA = ...;
+Key * keyB = ...;
+// assume that neither A below B nor B below A
+KeySet * a = ...;
+KeySet * b = ...;
+kdbGet(kdb, a, keyA); // (A)
+kdbGet(kdb, b, keyB); // (B)
+kdbSet(kdb, a, keyA); // (C)
+```
+
+- At point `(A)` the plugins internally store the KeySet for key `a`
+- At point `(B)` the plugins internally override the stored the KeySet with the KeySet for key `b`
+- At point `(C)` they will generate a wrong changeset for key `a`
+
+## Constraints
+
+- As far as possible, we must check for illegal sequences and raise an error when a constraint is violated
```

Comment:
Please stick to terminology, otherwise the sentences are becoming confusing. Give constraints a number so that you refer to it.

```suggestion
1. As far as possible, we must check for illegal sequences and raise an error when an illegal sequence occurs
```